### DISCUSSION TITLE
Release notes of the v17.11.20 ?

### DISCUSSION THREAD

**Discussion by GillesTourreau** [microsoft/vs-threading#1347] [General]
Hello,

Can we have a release notes of the changes/improvements of the v17.11.20 of the [Microsoft.VisualStudio.Threading.Analyzers](https://www.nuget.org/packages/Microsoft.VisualStudio.Threading.Analyzers/17.11.20) NuGet package ?

**Comment by AArnott** [comment]
Done: https://github.com/microsoft/vs-threading/releases

BTW, random usability question for you: when you look at the release notes do you review the notes from all releases (including prereleases) between your target and your baseline?
I'm wondering if folks expect a stable release to document all the changes since the last *stable* release vs. only since the last *prerelease*. Our policy so far is to only document changes since the nearest prior release (whether stable or prerelease).

**Comment by GillesTourreau** [comment]
Thanks @AArnott !

Personally, I always expected a release notes between changes for the last stable releases.
For the following main reasons:
- Most of the pre-release target advanced developers / testers. Just *5%* of the developers will use pre-release to test or discover new features (or bug fixes discovered in the previous last stable).
- A *standard developer* who use only the stable releases, will not check all the release notes of intermediate pre-releases (this is my opinion).
- The out-dated pre-release are often removed (unlisted in NuGet) to avoid developers to reference them and create issues with non-official releases (This is what I do on my public NuGet packages).
- Also, you can during different pre-releases add a new feature and remove it in the next pre-release (because not stable,...).
- Lot of developers don't know about the existence of the pre-release with NuGet package (with the *Include prerelease* check box in VS). And some developers don't want to use it, because they consider pre-release not stable / not official.

So for this reasons, me (but other developers I think) we don't check too much the delta releases of the pre-releases and read only the delta of the official releases in the release notes.
This is my opinion of course...

**Reply by AArnott** [reply]
Thank you for that. We may change our policy. But for now, be aware that the full list of release notes you're interested in can only be aggregated across _all_ the releases (including prereleases) between your base and target versions for this repo then.
